cmd/neighbors.go

```diff
@@ -6,12 +6,32 @@ import (
 	"net/http"
 	"os"
 
-	"github.com/kwhite17/Neighbors/pkg/database"
 	"github.com/kwhite17/Neighbors/pkg/managers"
 	"github.com/kwhite17/Neighbors/pkg/resources"
 	"github.com/kwhite17/Neighbors/pkg/retrievers"
+
+	"github.com/kwhite17/Neighbors/pkg/database"
+	"github.com/kwhite17/Neighbors/pkg/email"
+	"github.com/sendgrid/sendgrid-go"
+	"gopkg.in/gomail.v2"
 )
 
+type EnvironmentConfig struct {
+	Datasource  database.Datasource
+	EmailSender email.EmailSender
+}
+
+func BuildDatasource(driver string, host string, developmentMode bool) database.Datasource {
+	return database.BuildDatasource(driver, host, developmentMode)
+}
+
+func BuildEmailSender(developmentMode bool, userManager *managers.UserManager) email.EmailSender {
+	if developmentMode {
+		return &email.LocalSender{Dialer: &gomail.Dialer{Host: "localhost", Port: 25}, UserManager: userManager}
```

Comment:
You should consider using Mailtrap instead of local send, would remove a dependency for running this locally.

Reply:
i'm actually using smtp4dev: https://github.com/rnwood/smtp4dev